### PR TITLE
ARM64-SVE: Saturating*ByActiveElementCount

### PR DESCRIPTION
**Pull request by a74nh**

### CODE REVIEW

**Comment by a74nh** [comment]
Stress:
```
❯ ~/stress_tester.py $CORE_ROOT/corerun ./artifacts/tests/coreclr/linux.arm64.Checked/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/HardwareIntrinsics_Arm_ro.dll Sve_SaturatingDecrementByActiveElementCount
Starting test: /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/linux.arm64.Checked/Tests/Core_Root/corerun ./artifacts/tests/coreclr/linux.arm64.Checked/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/HardwareIntrinsics_Arm_ro.dll Sve_SaturatingDecrementByActiveElementCount
===================Running default===================
------------------- {} -------------------
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_int_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_int_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_int_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_int_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_long_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_long_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_long_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_long_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_uint_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_uint_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_uint_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_uint_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_ulong_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_ulong_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_ulong_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_ulong_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_vector_short() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_vector_int() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_vector_long() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_vector_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_vector_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_vector_ulong() : 7
===================Running jitstress===================
------------------- {'JitMinOpts': '1'} -------------------
------------------- {'JitStress': '1'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingDecrementByActiveElementCount<UInt64>(Vector<UInt64>, Vector<UInt64>): RunBasicScenario_Load failed:
    left: (9950131736413655575, 4465441822095806569)
   right: (0, 1)
  result: (9950131736413655575, 4465441822095806569)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingDecrementByActiveElementCount_vector_ulong() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingDecrementByActiveElementCount.vector.ulong.cs:line 62
   at Program.<<Main>$>g__TestExecutor3132|0_3133(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 78555
------------------- {'JitStress': '2'} -------------------
------------------- {'JitStress': '1', 'TieredCompilation': '1'} -------------------
------------------- {'JitStress': '2', 'TieredCompilation': '1'} -------------------
------------------- {'TailcallStress': '1'} -------------------
------------------- {'ReadyToRun': '0'} -------------------
===================Running jitstressregs===================
------------------- {'JitStressRegs': '1'} -------------------
------------------- {'JitStressRegs': '2'} -------------------
------------------- {'JitStressRegs': '3'} -------------------
------------------- {'JitStressRegs': '4'} -------------------
------------------- {'JitStressRegs': '8'} -------------------
------------------- {'JitStressRegs': '0x10'} -------------------
------------------- {'JitStressRegs': '0x80'} -------------------
------------------- {'JitStressRegs': '0x1000'} -------------------
------------------- {'JitStressRegs': '0x2000'} -------------------
===================Running jitstress2-jitstressregs===================
------------------- {'JitStress': '2', 'JitStressRegs': '1'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '2'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '3'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '4'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '8'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x10'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x80'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x1000'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '0x2000'} -------------------
```

```
❯ ~/stress_tester.py $CORE_ROOT/corerun ./artifacts/tests/coreclr/linux.arm64.Checked/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/HardwareIntrinsics_Arm_ro.dll Sve_SaturatingIncrementByActiveElementCount
Starting test: /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/linux.arm64.Checked/Tests/Core_Root/corerun ./artifacts/tests/coreclr/linux.arm64.Checked/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/HardwareIntrinsics_Arm_ro.dll Sve_SaturatingIncrementByActiveElementCount
===================Running default===================
------------------- {} -------------------
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_int_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_int_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_int_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_int_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_long_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_long_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_long_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_long_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_uint_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_uint_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_uint_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_uint_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_ulong_byte() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_ulong_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_ulong_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_ulong_ulong() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_short() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_long() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_ushort() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_uint() : 7
Passed test: _Sve_ro::JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_ulong() : 7
===================Running jitstress===================
------------------- {'JitMinOpts': '1'} -------------------
------------------- {'JitStress': '1'} -------------------
------------------- {'JitStress': '2'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (44330901, 80409478, 1977730174, 549934082)
   right: (1, 0, 1, 1)
  result: (44330901, 80409478, 1977730174, 549934082)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
------------------- {'JitStress': '1', 'TieredCompilation': '1'} -------------------
------------------- {'JitStress': '2', 'TieredCompilation': '1'} -------------------
------------------- {'TailcallStress': '1'} -------------------
------------------- {'ReadyToRun': '0'} -------------------
===================Running jitstressregs===================
------------------- {'JitStressRegs': '1'} -------------------
------------------- {'JitStressRegs': '2'} -------------------
------------------- {'JitStressRegs': '3'} -------------------
------------------- {'JitStressRegs': '4'} -------------------
------------------- {'JitStressRegs': '8'} -------------------
------------------- {'JitStressRegs': '0x10'} -------------------
------------------- {'JitStressRegs': '0x80'} -------------------
------------------- {'JitStressRegs': '0x1000'} -------------------
------------------- {'JitStressRegs': '0x2000'} -------------------
===================Running jitstress2-jitstressregs===================
------------------- {'JitStress': '2', 'JitStressRegs': '1'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (292303989, 378338709, 953996304, 798366385)
   right: (0, 0, 1, 0)
  result: (292303989, 378338709, 953996304, 798366385)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
------------------- {'JitStress': '2', 'JitStressRegs': '2'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '3'} -------------------
------------------- {'JitStress': '2', 'JitStressRegs': '4'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (819467066, 249429381, 2062736168, 1442643265)
   right: (1, 0, 1, 0)
  result: (819467066, 249429381, 2062736168, 1442643265)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
------------------- {'JitStress': '2', 'JitStressRegs': '8'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (951174538, 1165765203, 1616542767, 802986616)
   right: (1, 0, 1, 0)
  result: (951174538, 1165765203, 1616542767, 802986616)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
------------------- {'JitStress': '2', 'JitStressRegs': '0x10'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (249578136, 705899262, 1558623224, 409752869)
   right: (0, 1, 1, 0)
  result: (249578136, 705899262, 1558623224, 409752869)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
------------------- {'JitStress': '2', 'JitStressRegs': '0x80'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (717855585, 241806690, 78452716, 878696978)
   right: (0, 0, 1, 0)
  result: (717855585, 241806690, 78452716, 878696978)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
------------------- {'JitStress': '2', 'JitStressRegs': '0x1000'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (1862105264, 1666216739, 1862619847, 691182071)
   right: (1, 0, 1, 1)
  result: (1862105264, 1666216739, 1862619847, 691182071)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
------------------- {'JitStress': '2', 'JitStressRegs': '0x2000'} -------------------
Test failed:
..........................................
..........................................
Sve.SaturatingIncrementByActiveElementCount<Int32>(Vector<Int32>, Vector<Int32>): RunBasicScenario_Load failed:
    left: (90343983, 830076347, 711978603, 906125867)
   right: (0, 0, 0, 1)
  result: (90343983, 830076347, 711978603, 906125867)
..........................................
System.Exception: One or more scenarios did not complete as expected.
   at JIT.HardwareIntrinsics.Arm._Sve.Program.Sve_SaturatingIncrementByActiveElementCount_vector_int() in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/Arm/Sve/Sve_ro/Sve_ro/gen/Sve.SaturatingIncrementByActiveElementCount.vector.int.cs:line 62
   at Program.<<Main>$>g__TestExecutor3172|0_3173(StreamWriter tempLogSw, StreamWriter statsCsvSw, <>c__DisplayClass0_0&) in /home/alahay01/dotnet/runtime_sve_api/artifacts/tests/coreclr/obj/linux.arm64.Checked/Managed/JIT/HardwareIntrinsics/HardwareIntrinsics_Arm_ro/generated/XUnitWrapperGenerator/XUnitWrapperGenerator.XUnitWrapperGenerator/FullRunner.g.cs:line 79515
```

Need to check if these are known issues.

**Comment by a74nh** [comment]
@dotnet/arm64-contrib @kunalspathak

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp [hunk 1]
```diff
@@ -1926,6 +1926,32 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
...
+            {
+                // RMW semantics
+                if (targetReg != op1Reg)
+                {
+                    assert(targetReg != op2Reg);
+                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
+                }
+
+                // Switch instruction if arg1 is unsigned.
+                if (varTypeIsUnsigned(node->GetAuxiliaryType()))
```
Comment (kunalspathak) [inline src/coreclr/jit/hwintrinsiccodegenarm64.cpp:1940]: shouldn't this also use the unsigned variant of the instruction i.e. the first arg is `Vector<ulong>` and hence, the base type is `ulong`.

```c#
public static System.Numerics.Vector<ulong> SaturatingDecrementByActiveElementCount(System.Numerics.Vector<ulong> value, System.Numerics.Vector<ulong> from) { throw null; }
```

**Review by kunalspathak — COMMENTED** [review]
Looks good, except a question.

File: src/coreclr/jit/hwintrinsiccodegenarm64.cpp (hunk 1 above)
Comment (a74nh) [reply src/coreclr/jit/hwintrinsiccodegenarm64.cpp:1940]: Aux type is set as:
`retNode->AsHWIntrinsic()->SetAuxiliaryJitType(op1BaseJitType);`

So, for the example, `GetAuxiliaryType()` should be returning `ulong`

Comment (kunalspathak) [reply src/coreclr/jit/hwintrinsiccodegenarm64.cpp:1940]: ah, right.

**Review by kunalspathak — APPROVED** [review]
LGTM. Can you fix the formatting errors too?

**Comment by kunalspathak** [comment]
@a74nh - there is a merge conflict.

File: src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs [hunk 2]
```diff
@@ -6655,6 +6655,85 @@ public static int NumberOfElementsInVectorInt64(SveMaskPattern pattern)
...
+
+        public static int NumberOfActiveElementsInMask(ulong[] mask)
+        {
+            int acc = 0;
+            for (var i = 0; i < mask.Length; i++)
+            {
+                acc += (mask[i] == 1) ? 1 : 0;
+            }
+            return acc;
+
```
Comment (kunalspathak) [inline src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs:6736]: Missing `}` is causing build failure.

Comment (a74nh) [reply src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs:6737]: Fixed :)

**Comment by kunalspathak** [comment]
/ba-g superpmi-replay failures are #102773 and machine agent issue

**Pull request merged**